src/autogpt_plugins/email/email_plugin/email_plugin.py

```diff
@@ -311,3 +312,58 @@ def clean_email_body(email_body):
     email_body = re.sub(r"http\S+", "", email_body)
 
     return email_body
+
+def truncate_emails(messages):
+    """Truncate email content based on configuration settings
+
+    Args:
+        messages (messages, optional): List of emails
+
+    Returns:
+        messages: List of truncated emails
+    """
+
+    # If MAX_CHARACTERS is breached
+    messages_length = len(str(messages))
+    if os.getenv("MAX_CHARACTERS") and int(os.getenv("MAX_CHARACTERS")) < int(messages_length):
+
+        # If MAX_NUM_EMAILS is breached, return the maximum number of emails
+        if os.getenv("MAX_NUM_EMAILS") and int(os.getenv("MAX_NUM_EMAILS")) < int(len(messages)):
+            end_index = len(messages) - 1
+            start_index = len(messages) - int(os.getenv("MAX_NUM_EMAILS"))
+
+            return messages[start_index:end_index]
+
+        # If MAX_CHARACTERS is breached while MAX_NUM_EMAILS is not breached, truncate email body's
+        truncation_message = " [Warning: This email's body was truncated by the Auto-GPT email plugin to avoid \"command returned too much output errors\". Retrieve fewer emails to get the full message body]"
```

Comment:
This warning message also takes up token space. We should make it very concise e.g.:
`[... Email truncated]`